src/mds/MDSRank.cc

```diff
@@ -3847,3 +3854,27 @@ void MDSRank::send_task_status() {
 
   schedule_update_timer_task();
 }
+
+void MDSRank::schedule_inmemory_logger() {
+  timer.add_event_after(g_conf().get_val<double>("mds_extraordinary_events_dump_interval"),
+                        new LambdaContext([this]() {
+                          inmemory_logger();
+                        }));
+}
+
+void MDSRank::inmemory_logger() {
+  if (g_conf().get_val<double>("mds_extraordinary_events_dump_interval") &&
+      (client_eviction_dump || beacon.missed_beacon_ack_dump ||
+       beacon.missed_internal_heartbeat_dump || state < MDSMap::STATE_ACTIVE)) {
+    //dump the in-memory logs if any of these events occured recently
+    g_ceph_context->_log->dump_recent();
+    reset_event_flags();
+  }
+  schedule_inmemory_logger();
```

Comment:
IMO you should skip scheduling this if the `mds_extraordinary_events_dump_interval` option is disabled. No need to loop this which makes no sense.

And try to trigger this in `reset_event_flags()` when the option is enabled.

Reply:
And this will also need to take the global `mds_lock` too for each timer.

Reply:
> IMO you should skip scheduling this if the `mds_extraordinary_events_dump_interval` option is disabled. No need to loop this which makes no sense.
> 
> And try to trigger this in `reset_event_flags()` when the option is enabled.

The original implementation was like that. But I've changed it because it failed to schedule again when we set `mds_extraordinary_events_dump_interval`. Will try that once again and check.